--- common/autoconf/basics.m4.orig
+++ common/autoconf/basics.m4
@@ -555,10 +555,12 @@ AC_DEFUN([BASIC_CHECK_GNU_MAKE],
   # We need to find a recent version of GNU make. Especially on Solaris, this can be tricky.
   if test "x$MAKE" != x; then
     # User has supplied a make, test it.
-    if test ! -f "$MAKE"; then
-      AC_MSG_ERROR([The specified make (by MAKE=$MAKE) is not found.])
+    if test `dirname "$MAKE"` = "." && ! test -f "$MAKE"; then
+      AC_PATH_PROGS(CHECK_MAKE, $MAKE)
+    else
+      CHECK_MAKE="${MAKE}"
     fi
-    BASIC_CHECK_MAKE_VERSION("$MAKE", [user supplied MAKE=$MAKE])
+    BASIC_CHECK_MAKE_VERSION("$CHECK_MAKE", [user supplied MAKE=$MAKE])
     if test "x$FOUND_MAKE" = x; then
       AC_MSG_ERROR([The specified make (by MAKE=$MAKE) is not GNU make 3.81 or newer.])
     fi
