From 27c44b052ca69873046f32631b3ccbe1d6b11dd3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andreas=20M=C3=BCller?= <schnitzeltony@googlemail.com>
Date: Sun, 15 Mar 2015 00:57:10 +0100
Subject: [PATCH] use pkg-config to gather sdl information
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Upstream-Status: Inappropriate [oe specific]

Signed-off-by: Andreas Müller <schnitzeltony@googlemail.com>
---
 configure | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/configure b/configure
index 9496cf8..1bce86e 100755
--- a/configure
+++ b/configure
@@ -2710,13 +2710,12 @@ MODULES="$MODULES backends/platform/$_backend"
 #
 case $_backend in
 	dingux | gp2x | gph | linuxmoto | maemo | openpandora | samsungtv | sdl)
-		find_sdlconfig
-		INCLUDES="$INCLUDES `$_sdlconfig --prefix="$_sdlpath" --cflags`"
-		LIBS="$LIBS `$_sdlconfig --prefix="$_sdlpath" --libs`"
+		INCLUDES="$INCLUDES `pkg-config --cflags sdl`"
+		LIBS="$LIBS `pkg-config --libs sdl`"
 		DEFINES="$DEFINES -DSDL_BACKEND"
 		add_line_to_config_mk "SDL_BACKEND = 1"
 
-		_sdlversion=`$_sdlconfig --version`
+		_sdlversion=`pkg-config --modversion`
 		case $_sdlversion in
 			1.3.*)
 				add_line_to_config_mk "USE_SDL13 = 1"
-- 
1.9.3

