From 8771edf61fcf3ced4afb9e9bc08897f575d86834 Mon Sep 17 00:00:00 2001
From: Deve <deveee@gmail.com>
Date: Sat, 17 May 2014 09:18:25 +0200
Subject: [PATCH 1/2] Use system's glext.h/glxext.h on linux.

---
 lib/irrlicht/source/Irrlicht/COpenGLExtensionHandler.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/irrlicht/source/Irrlicht/COpenGLExtensionHandler.h b/lib/irrlicht/source/Irrlicht/COpenGLExtensionHandler.h
index 7f9e0df..586d0c4 100644
--- a/lib/irrlicht/source/Irrlicht/COpenGLExtensionHandler.h
+++ b/lib/irrlicht/source/Irrlicht/COpenGLExtensionHandler.h
@@ -61,9 +61,9 @@
 	#include <GL/gl.h>
 	#include <GL/glx.h>
 	#if defined(_IRR_OPENGL_USE_EXTPOINTER_)
-	#include "glext.h"
+	#include <GL/glext.h>
 	#undef GLX_ARB_get_proc_address // avoid problems with local glxext.h
-	#include "glxext.h"
+	#include <GL/glxext.h>
 	#endif
 #endif
 
-- 
2.1.4

