Give server more time to start

Signed-off-by: Marko Lindqvist <cazfi74@gmail.com>

diff -Nurd -X.diff_ignore freeciv/client/connectdlg_common.c freeciv/client/connectdlg_common.c
--- freeciv/client/connectdlg_common.c	2016-04-25 15:07:18.840145337 +0300
+++ freeciv/client/connectdlg_common.c	2016-05-04 09:43:11.524504302 +0300
@@ -65,7 +65,7 @@
 #include "tilespec.h"
 
 #define WAIT_BETWEEN_TRIES 100000 /* usecs */ 
-#define NUMBER_OF_TRIES 500
+#define NUMBER_OF_TRIES 10000
 
 #if defined(HAVE_WORKING_FORK) && !defined(WIN32_NATIVE)
 /* We are yet to see WIN32_NATIVE setup where even HAVE_WORKING_FORK would
