From 4cd511613675220a3fdb98d58bfb558d7eacc754 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andreas=20M=C3=BCller?= <schnitzeltony@googlemail.com>
Date: Sat, 14 Mar 2015 00:48:57 +0100
Subject: [PATCH] find libgl and libglu by pkg-config
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Upstrema-Status: pending

Signed-off-by: Andreas Müller <schnitzeltony@googlemail.com>
---
 configure.ac | 16 ++++++++++------
 1 file changed, 10 insertions(+), 6 deletions(-)

diff --git a/configure.ac b/configure.ac
index 7dd6817..8af8a59 100644
--- a/configure.ac
+++ b/configure.ac
@@ -63,16 +63,20 @@ PKG_CHECK_MODULES([SDLMIXER], [SDL_mixer],
 ],
 [AC_MSG_ERROR([SDL_mixer not found])])
 
-AC_CHECK_LIB([GL], [glBegin], [],
+PKG_CHECK_MODULES([GL], [gl],
+[
+  CFLAGS="${CFLAGS} $GL_CFLAGS"
+  LIBS="${LIBS} $GL_LIBS"
+],
 [AC_MSG_ERROR([libGL not found])])
 
-LIBS="$LIBS -lGL"
-
-AC_CHECK_LIB([GLU], [gluBeginCurve], [],
+PKG_CHECK_MODULES([GLU], [glu],
+[
+  CFLAGS="${CFLAGS} $GLU_CFLAGS"
+  LIBS="${LIBS} $GLU_LIBS"
+],
 [AC_MSG_ERROR([libGLU not found])])
 
-LIBS="$LIBS -lGLU"
-
 CPPFLAGS="${CPPFLAGS} -DETR_DATA_DIR=\\\"$datadir\\\""
 
 # Request c++11 compatibility
-- 
1.9.3

