diff -Nurd etr-0.6.0/configure.ac etr-0.6.0-newbie/configure.ac
--- etr-0.6.0/configure.ac	2013-09-13 22:39:29.000000000 +0300
+++ etr-0.6.0-newbie/configure.ac	2014-08-29 08:13:30.094868910 +0300
@@ -15,10 +15,6 @@
 AC_PROG_CC
 
 # Checks for libraries.
-AC_CHECK_FT2
-
-CXXFLAGS="$CXXFLAGS $FT2_CFLAGS"
-LIBS="$LIBS $FT2_LIBS"
 
 # Checks for header files.
 AC_CHECK_HEADERS([float.h limits.h stdlib.h string.h sys/time.h unistd.h])
@@ -46,6 +42,13 @@
 CFLAGS="$CFLAGS $SDL_CFLAGS"
 LIBS="$LIBS $SDL_LIBS"
 
+PKG_CHECK_MODULES([FT2], [freetype2],
+[
+  CXXFLAGS="$CXXFLAGS $FT2_CFLAGS"
+  LIBS="$LIBS $FT2_LIBS"
+],
+[AC_MSG_ERROR([Freetype not found])])
+
 PKG_CHECK_MODULES([SDLIMAGE], [SDL_image],
 [
   CFLAGS="${CFLAGS} $SDLIMAGE_CFLAGS"
